prime-router/docs/design/proposals/mapping-inventory-test-plan.md

```diff
@@ -0,0 +1,50 @@
+be able to transform valid NIST 2.5.1 messages losslessly to FHIR using the official HL7v2 mapping inventory as a guide. 
+UP shall be able to transform FHIR ELR messages to valid NIST 2.5.1."
+
+Currently, we have integration tests that test an unknown combination of parts of sender transforms, receiver 
+transforms, and mapping inventory rules. We want a clear and definitive way of knowing what is already tested, and we 
+want the mapping inventory testing to be thorough. This means testing not only that each piece of the message ends up 
+in the Primary Target (as defined in column J of the mapping inventory spreadsheet), but also that the conditions set
+forth in the Computable ANTLR column of the mapping inventory spreedsheet are tested. This will just test moving from 
+HL7 V2 -> FHIR. We also need to test that the resulting FHIR message can be converted back to HL7 V2 losslessly. To 
+achieve this, we can reverse the order of the files and ensure that the output from processing the FHIR file matches 
+the HL7 V2 file. This will help keep the number of files needed down. 
+
+Integration tests are currently rather challenging to read. At a minimum, we need a standard for naming the files. When
+testing mapping inventory segments, we will name files with the format `segment_#`. An example, 
+`PID_1`. The corresponding FHIR file will be named the same, but will have a `.fhir` extension. This way, it is 
+very apparent not only what the file is testing, but which file goes with it. If it gets down to where there may only 
+be one complicated Computable ANTLR rule being tested, the field should be specified as well since there is only one 
+thing being tested at that point. ex. `PID_patient_name_5`.  
+
+There will be exceptions to this. There are datatypes that are within the primary segments, for instance `XPN`. 
+We don't want to test these repeatedly for each segment they appear in, so we will have separate files for each of them
+labeled like so `XPN_1`. Each file should have a comment description at the top specifying exactly what is tested in 
+that file, since each file will be responsible for testing each piece of the segment.
+
+Instead of piling these files into the existing file structure,
+especially since we will be using them for multidirectional tests, we will put them in a folder structure 
+`datatests/mapping-inventory/segment-name`.
+
+Within the files, we will only put the bare minimum number of segments in a file. `MSH` is still required, so, for 
+instance, when testing `PID` you would only have an `MSH` segment and a `PID` segment in the file. To allow this, when 
+converting form the HL7 V2 file to the FHIR file, use the new test schema 
+`metadata/hl7_mapping/testing/ORU_R01-test.yml`
+
+`translation-test-config.csv` is where our current tests like this live. We will create a separate file called 
+`mapping-inventory-test-config.csv` since this file is already bloated, difficult to read, and lacks organization. 
```

Comment:
Circling back here, I'm wondering if continuing with this CSV is the right approach since it will also become super bloated and it makes it very difficult to understand what a test is actually covering.

There is also the logistical downside of the dynamic tests in that you cannot run a single test case without manually editing the CSV>